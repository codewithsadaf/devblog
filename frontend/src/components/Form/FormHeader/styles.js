//imports
import styled, { css } from "styled-components";
//components styles
import { TextComponent } from "../../Layout/TextComponent";

export const Container = styled.div`
    ${({ theme }) => css`
        margin-bottom: 1em;
    `}
`;

export const Text = styled(TextComponent)`
    margin-top: 0.3em;
`;
