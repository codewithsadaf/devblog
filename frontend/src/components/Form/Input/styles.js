//imports
import styled, { css } from "styled-components";

export const FormControl = styled.div`
    display: flex;
    flex-direction: column;
    margin-bottom: 1em;
`;

export const LabelComponent = styled.label`
    display: block;
    margin-bottom: 0.3em;
    font-weight: 700;
`;

export const InputComponent = styled.input`
    ${({ theme }) => css`
        padding: 7px 0;
        outline: none;
        border: none;
        border-bottom: 1px solid ${theme.colors.gray};
        background-color: transparent;
        font-size: ${theme.font.sizes.default};
        transition: ${theme.transition.tran_05};

        &:focus {
            border-bottom: 1px solid ${theme.colors.primaryColor};
            transition: ${theme.transition.tran_03};
        }
    `}
`;
