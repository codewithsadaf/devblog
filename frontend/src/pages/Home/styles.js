//imports
import styled, { css } from "styled-components";

export const HomeContainer = styled.div`
    ${({ theme }) => css`
        width: 600px;
        margin: 3em auto 0;

        @media ${theme.media.lteMedium} {
            max-width: 100%;
        }
    `}
`;

export const MessageWrapper = styled.div`
    margin-top: 5em;
    text-align: center;
    display: flex;
    align-items: center;
    justify-content: center;
`;
