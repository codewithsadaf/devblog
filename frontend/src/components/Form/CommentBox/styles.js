//imports
import styled, { css } from "styled-components";
//components styles
import { InputComponent } from "../Input/styles";
import { ButtonContainer as Button } from "../../Layout/Button/styles";

export const Container = styled.div`
    display: flex;
    align-items: center;
    justify-content: space-between;
    width: 100%;

    & ${Button} {
        padding: 12px 16px;
    }
`;

export const CommentInput = styled(InputComponent)`
    ${({ theme }) => css`
        border: 1px solid ${theme.colors.gray};
        transition: ${theme.transition.tran_03};
        padding: 12px 7px;
        width: 90%;
        border-radius: 5px;
        font-size: 1em;
        margin-right: 7px;

        &:focus {
            border: 1px solid ${theme.colors.primaryColor};
            transition: ${theme.transition.tran_03};
        }
    `}
`;
