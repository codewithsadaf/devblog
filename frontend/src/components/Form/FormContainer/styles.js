//imports
import styled, { css } from "styled-components";
//components styles
import { ButtonContainer as Button } from "../../../components/Layout/Button/styles";

export const Container = styled.div`
    width: 100%;
    height: 100%;
    display: flex;
    align-items: center;
    justify-content: center;
`;

export const Form = styled.form`
    ${({ theme }) => css`
        width: 450px;
        margin-top: 3em;

        & > ${Button} {
            margin-top: 1em;
        }

        @media ${theme.media.lteMedium} {
            width: 100%;
            min-width: 0;
            max-width: 100%;
        }
    `}
`;
