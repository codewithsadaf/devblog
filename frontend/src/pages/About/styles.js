//imports
import styled, { css } from "styled-components";
//components styles
import { Text as TextComponent } from "../../components/Layout/TextComponent/styles";
import { Container as Heading } from "../../components/Layout/Heading/styles";

export const AboutContainer = styled.div`
    ${({ theme }) => css`
        max-width: 60%;
        margin: 3em auto 0;
        text-align: center;

        & > ${TextComponent} {
            margin-top: 0.5em;
        }

        @media ${theme.media.lteMedium} {
            max-width: 100%;

            & > ${Heading} {
                font-size: ${theme.font.sizes.xxlarge};
                line-height: ${theme.font.line_height.h3};
            }
        }
    `}
`;
