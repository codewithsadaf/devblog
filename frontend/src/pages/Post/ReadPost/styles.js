//imports
import styled, { css } from "styled-components";
import { Link } from "react-router-dom";
//components styles
import { ImageElm } from "../../../components/Layout/ImageBox/styles";
import { Text as TextComponent } from "../../../components/Layout/TextComponent/styles";

//PAGE CONTAINER
export const Container = styled.div`
    margin: 3em auto 0;
`;

//RETURN TO HOME TEXT
export const BackWrapper = styled.div`
    width: 100%;
    display: flex;
    align-items: center;
    justify-content: flex-start;
    margin-bottom: 1.2em;
    align-items: center;
`;

export const BackLink = styled(Link)`
    ${({ theme }) => css`
        color: ${theme.colors.primaryColor};
        transition: ${theme.transition.tran_03};
        display: flex;
        align-items: center;

        &:hover {
            transition: ${theme.transition.tran_03};
            opacity: 0.7;
        }
    `}
`;

//POST HEADER
export const PostHeader = styled.div`
    ${({ theme }) => css`
        width: 80%;
        margin: 0 auto;
        text-align: center;

        ${ImageElm} {
            height: 50%;
            width: 50%;
            margin: 2em auto;

            @media ${theme.media.lteMedium} {
                width: 100%;
                height: 100%;
            }
        }

        ${TextComponent} {
            margin: 1em 0;
        }

        @media ${theme.media.lteMedium} {
            width: 100%;
        }
    `}
`;

//POST CONTENT
export const ContentWrapper = styled.div`
    ${({ theme }) => css`
        width: 60%;
        margin: 0 auto;

        @media ${theme.media.lteMedium} {
            width: 100%;
        }
    `}
`;

export const DisplayContent = styled.div`
    ${({ theme }) => css`
        text-align: left;

        h1,
        h2,
        h3,
        h4,
        h5,
        h6 {
            margin-bottom: 0.5em;
            font-weight: bold;
        }

        p {
            font-size: 18px;
            margin-bottom: 1em;
        }

        a {
            font-weight: bold;
            color: ${theme.colors.primaryColor};
            text-decoration: underline;
        }

        ol,
        ul {
            margin-left: 2em;

            li {
                margin-bottom: 1em;
            }
        }

        ul {
            list-style-type: disc;
        }

        ol {
            list-style-type: decimal;
        }

        img {
            width: 100%;
            display: block;
            margin: 2em 0;
        }
    `}
`;

//POST TAGS
export const TagsWrapper = styled.div`
    display: flex;
    justify-content: center;
    align-items: center;
    width: 100%;
    margin: 3em 0;
    flex-wrap: wrap;
`;

export const TagText = styled.span`
    display: block;
    padding: 0 0.5em;
    font-weight: 500;

    &::before {
        content: "#";
        font-weight: bold;
    }
`;

export const TagsInfo = styled.span`
    display: block;
    margin-right: 0.5em;
    font-weight: bold;
`;

//POST LINE
export const Line = styled.div`
    ${({ theme }) => css`
        height: 2px;
        width: 100%;
        background-color: ${theme.colors.footerBackground};
        margin: 1em 0;
    `}
`;

//POST COMMENTS
export const CommentsNumberText = styled.h5`
    ${({ theme }) => css`
        font-size: ${theme.font.sizes.large};
        line-height: ${theme.font.line_height.h5};
    `}
`;

export const CommentsWrapper = styled.div`
    width: 100%;
    display: flex;
    flex-direction: column;
`;

export const SignUpWrapper = styled.div`
    margin: 2em 0;
    display: flex;
    align-items: center;
    justify-content: space-between;
    flex-direction: column;
    width: 100%;
    text-align: center;

    button {
        margin-top: 1em;
    }
`;

export const SignUpText = styled.h4`
    ${({ theme }) => css`
        font-size: ${theme.font.sizes.xlarge};
        line-height: ${theme.font.line_height.h4};
    `}
`;
