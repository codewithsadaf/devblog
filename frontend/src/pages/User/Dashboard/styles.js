//imports
import styled, { css } from "styled-components";
//components styles
import { ImageElm } from "../../../components/Layout/ImageBox/styles";
import { Container as Heading } from "../../../components/Layout/Heading/styles";
import { ButtonContainer as Button } from "../../../components/Layout/Button/styles";
import { InputComponent } from "../../../components/Form/Input/styles";
//react icons
import { FaCog } from "react-icons/fa";
import { Link } from "react-router-dom";

export const Container = styled.div`
    ${({ theme }) => css`
        width: 80%;
        margin: 3em auto 0;

        @media ${theme.media.lteMedium} {
            width: 100%;
        }
    `}
`;

export const TitleWrapper = styled.div`
    display: flex;
    align-items: center;
    justify-content: center;
    width: 100%;
    margin-bottom: 1.2em;
`;

export const ProfileWrapper = styled.div`
    ${({ theme }) => css`
        display: flex;
        align-items: center;
        justify-content: space-between;

        @media ${theme.media.lteMedium} {
            flex-direction: column;
        }
    `}
`;

export const ProfileImageWrapper = styled.div`
    ${({ theme }) => css`
        flex-basis: 40%;

        ${ImageElm} {
            width: 300px;
            height: 300px;
            object-fit: fill;
            border-radius: 50%;

            @media ${theme.media.lteMedium} {
                width: 150px;
                height: 150px;
            }
        }

        @media ${theme.media.lteMedium} {
            flex-basis: 100%;
        }
    `}
`;

export const ProfileInformationWrapper = styled.div`
    ${({ theme }) => css`
        flex-basis: 58%;
        min-height: 250px;
        text-align: left;

        ${Heading}:nth-of-type(2) {
            margin: 0.3em 0;
        }

        @media ${theme.media.lteMedium} {
            flex-basis: 100%;
            width: 100%;
            margin-left: 0;
        }
    `}
`;

export const RowStart = styled.div`
    display: flex;
    align-items: center;
    justify-content: flex-start;
    flex-wrap: wrap;
`;

export const CogLink = styled(Link)`
    ${({ theme }) => css`
        font-size: 1.2em;
        margin-left: auto;
        cursor: pointer;
        color: ${theme.colors.primaryColor};
        opacity: 1;
        transition: ${theme.transition.tran_05};

        &:hover {
            opacity: 0.6;
            transition: ${theme.transition.tran_05};
        }
    `}
`;

export const CogIcon = styled(FaCog)`
    ${({ theme }) => css``}
`;

export const PostsCommentsInfoWrapper = styled.div`
    ${({ theme }) => css`
        display: flex;
        width: 100%;
        border: 1px solid ${theme.colors.gray};
        border-radius: 7px;
        padding: 20px;
        margin-top: 1em;
    `}
`;

export const PostsInfoWrapper = styled.div`
    display: flex;
    flex-direction: column;
    align-items: center;
    margin-right: 2em;
`;

export const CommentsInfoWrapper = styled.div`
    display: flex;
    flex-direction: column;
    align-items: center;
`;

export const ViewOptionWrapper = styled.div`
    ${({ theme }) => css`
        margin: 2em 0;
        display: flex;
        border-bottom: 2px solid ${theme.colors.primaryColor};
    `}
`;

export const ViewOption = styled.span`
    ${({ theme }) => css`
        background-color: transparent;
        color: ${theme.colors.primaryColor};
        padding: 7px;
        font-size: ${theme.font.sizes.medium};
        border-radius: 7px 7px 0 0;
        font-weight: 500;
        cursor: pointer;
        transition: ${theme.transition.tran_03};

        &.active {
            background-color: ${theme.colors.primaryColor};
            color: ${theme.colors.white};
            transition: ${theme.transition.tran_03};
            pointer-events: none;
        }
    `}
`;

export const DashboardHeader = styled.div`
    ${({ theme }) => css`
        display: flex;
        justify-content: space-between;
        align-items: center;
        width: 100%;
        border-bottom: 2px solid ${theme.colors.gray};
        font-weight: 700;
        padding: 10px;
    `}
`;

export const DashboardText = styled.span`
    ${({ theme }) => css`
        font-size: ${theme.font.sizes.default};
        line-height: ${theme.font.line_height.default};
    `}
`;

export const DashboardRow = styled.div`
    ${({ theme }) => css`
        display: flex;
        justify-content: space-between;
        align-items: center;
        padding: 10px;
        width: 100%;
        border-bottom: 2px solid ${theme.colors.lightGray};

        & ${DashboardText} {
            display: block;
            width: 60%;
            word-wrap: break-word;

            @media ${theme.media.lteMedium} {
                width: 100%;
            }
        }

        @media ${theme.media.lteMedium} {
            flex-direction: column;
            justify-content: flex-start;
            align-items: flex-start;
        }
    `}
`;

export const DashboardActionsWrapper = styled.div`
    ${({ theme }) => css`
        margin-left: 2em;
        display: flex;
        justify-content: space-between;
        align-items: center;

        & ${Button} {
            margin-left: 0.5em;

            @media ${theme.media.lteMedium} {
                margin-left: 0;
                margin-right: 0.5em;
            }
        }

        @media ${theme.media.lteMedium} {
            margin-left: 0;
            margin: 1em 0;
        }
    `}
`;

export const CenterTextWrapper = styled.div`
    display: flex;
    align-items: center;
    justify-content: center;
    width: 100%;
    text-align: center;
    padding: 2em 0;
`;

export const EditInput = styled(InputComponent)`
    border: none;
    width: 100%;

    &:focus {
        border: none;
    }
`;
