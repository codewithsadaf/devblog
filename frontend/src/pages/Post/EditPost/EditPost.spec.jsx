//imports
import React from "react";
import { fireEvent, screen, waitFor } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import { act } from "react-dom/test-utils";
//styles
import { renderTheme } from "../../../styles/render-theme";
//contexts
import { Context as UserContext } from "../../../context/UserContext";
import { PostContext } from "../../../context/PostContext";
//components
import { EditPost } from "./index";

//setting scrollTo mock
const spyScrollTop = jest.fn();
Object.defineProperty(window.global, "scrollTo", { value: spyScrollTop });

//post data mock
const mockPostData = {
    image: "example.png",
    title: "post title",
    content: "<p>post content</p>",
    tags: ["#tag1", "#tag2"],
};

describe("<EditPost />", () => {
    //user context values mock
    let mockAuthenticated;
    //post context values mock
    let mockIsLoading;
    let mockEditPost;
    let mockGetPostById;

    beforeEach(() => {
        mockAuthenticated = true;
        mockIsLoading = false;
        mockEditPost = jest.fn();
        mockGetPostById = jest.fn();

        //returning mock promise from mockGetPostById
        mockGetPostById.mockReturnValue(Promise.resolve(mockPostData));
    });

    describe("Rendering", () => {
        it("should render EditPost Component with default values", async () => {
            renderTheme(
                <BrowserRouter>
                    <UserContext.Provider
                        value={{
                            authenticated: mockAuthenticated,
                        }}
                    >
                        <PostContext.Provider
                            value={{
                                isLoading: mockIsLoading,
                                EditPost: mockEditPost,
                                getPostById: mockGetPostById,
                            }}
                        >
                            <EditPost />
                        </PostContext.Provider>
                    </UserContext.Provider>
                </BrowserRouter>
            );

            //wait for mockGetPostById to have been called before proceeding
            await waitFor(() => {
                expect(mockGetPostById).toHaveBeenCalledTimes(1);
            });

            const imageInput = document.querySelector('input[name="image"]');
            const titleInput = document.querySelector('input[name="title"]');
            const contentEditor = document.querySelector(".ql-editor");
            const tagsInput = document.querySelector('input[name="tags"]');
            const EditPostButton = screen.getByRole("button", {
                name: "Edite seu post",
            });

            expect(imageInput).toBeInTheDocument();
            expect(titleInput).toBeInTheDocument();
            expect(contentEditor).toBeInTheDocument();
            expect(tagsInput).toBeInTheDocument();
            expect(EditPostButton).toBeInTheDocument();

            //default values
            expect(
                screen.getByRole("img", { name: mockPostData.title })
            ).toHaveAttribute("src", mockPostData.image);
            expect(titleInput.value).toBe(mockPostData.title);
            expect(contentEditor.querySelector("p").innerHTML).toBe(
                mockPostData.content.replace("<p>", "").replace("</p>", "")
            );
            expect(tagsInput.value).toBe(mockPostData.tags.join(" "));
        });

        it("should render PostForm with headingText and subHeadingText correct value", async () => {
            renderTheme(
                <BrowserRouter>
                    <UserContext.Provider
                        value={{
                            authenticated: mockAuthenticated,
                        }}
                    >
                        <PostContext.Provider
                            value={{
                                isLoading: mockIsLoading,
                                EditPost: mockEditPost,
                                getPostById: mockGetPostById,
                            }}
                        >
                            <EditPost />
                        </PostContext.Provider>
                    </UserContext.Provider>
                </BrowserRouter>
            );

            //wait for mockGetPostById to have been called before proceeding
            await waitFor(() => {
                expect(mockGetPostById).toHaveBeenCalledTimes(1);
            });

            const heading = screen.getByRole("heading", {
                name: "Editar Post",
            });
            const subHeading = screen.getByText("Revise e atualize o seu post");

            expect(heading).toBeInTheDocument();
            expect(subHeading).toBeInTheDocument();
        });

        it("should not render EditPost component when authenticated is false", () => {
            renderTheme(
                <BrowserRouter>
                    <UserContext.Provider
                        value={{
                            authenticated: false,
                        }}
                    >
                        <PostContext.Provider
                            value={{
                                isLoading: mockIsLoading,
                                EditPost: mockEditPost,
                                getPostById: mockGetPostById,
                            }}
                        >
                            <EditPost />
                        </PostContext.Provider>
                    </UserContext.Provider>
                </BrowserRouter>
            );
            const postForm = screen.queryByLabelText("post-form");
            const EditPostButton = screen.queryByRole("button", {
                name: "Edite seu post",
            });

            expect(postForm).not.toBeInTheDocument();
            expect(EditPostButton).not.toBeInTheDocument();
        });

        it("should render loading when isLoading is true", async () => {
            renderTheme(
                <BrowserRouter>
                    <UserContext.Provider
                        value={{
                            authenticated: mockIsLoading,
                        }}
                    >
                        <PostContext.Provider
                            value={{
                                isLoading: true,
                                EditPost: mockEditPost,
                                getPostById: mockGetPostById,
                            }}
                        >
                            <EditPost />
                        </PostContext.Provider>
                    </UserContext.Provider>
                </BrowserRouter>
            );

            expect(screen.getByLabelText("Loader-wrapper")).toBeInTheDocument();
        });

        it("should match snapshot", async () => {
            const { container } = renderTheme(
                <BrowserRouter>
                    <UserContext.Provider
                        value={{
                            authenticated: mockAuthenticated,
                        }}
                    >
                        <PostContext.Provider
                            value={{
                                isLoading: mockIsLoading,
                                EditPost: mockEditPost,
                                getPostById: mockGetPostById,
                            }}
                        >
                            <EditPost />
                        </PostContext.Provider>
                    </UserContext.Provider>
                </BrowserRouter>
            );

            //wait for mockGetPostById to have been called before proceeding
            await waitFor(() => {
                expect(mockGetPostById).toHaveBeenCalledTimes(1);
            });

            expect(container.firstChild).toMatchSnapshot();
        });
    });
    describe("Functionality", () => {
        it("should call edit post function on button click ", async () => {
            renderTheme(
                <BrowserRouter>
                    <UserContext.Provider
                        value={{
                            authenticated: mockAuthenticated,
                        }}
                    >
                        <PostContext.Provider
                            value={{
                                isLoading: mockIsLoading,
                                editPost: mockEditPost,
                                getPostById: mockGetPostById,
                            }}
                        >
                            <EditPost />
                        </PostContext.Provider>
                    </UserContext.Provider>
                </BrowserRouter>
            );

            //wait for mockGetPostById to have been called before proceeding
            await waitFor(() => {
                expect(mockGetPostById).toHaveBeenCalledTimes(1);
            });

            const EditPostButton = screen.getByRole("button", {
                name: "Edite seu post",
            });

            act(() => {
                fireEvent.click(EditPostButton);
            });

            expect(mockEditPost).toHaveBeenCalledTimes(1);
        });
    });
});
