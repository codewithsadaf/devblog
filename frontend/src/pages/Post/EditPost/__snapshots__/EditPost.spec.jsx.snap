// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditPost /> Rendering should match snapshot 1`] = `
.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-bottom: 1em;
}

.c7 {
  display: block;
  margin-bottom: 0.3em;
  font-weight: 700;
}

.c8 {
  padding: 7px 0;
  outline: none;
  border: none;
  border-bottom: 1px solid #ccc;
  background-color: transparent;
  font-size: 1em;
  -webkit-transition: all 500ms ease-in-out;
  transition: all 500ms ease-in-out;
}

.c8:focus {
  border-bottom: 1px solid #000;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: 1em 0;
}

.c10 {
  display: block;
  margin-bottom: 0.3em;
  font-weight: 700;
}

.c12 {
  padding: 0.5em 1.8em;
  font-size: 1em;
  border: none;
  font-weight: bold;
  cursor: pointer;
  background: #04a453;
  color: #fff;
  border: 2px solid #04a453;
  border-radius: 4px;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
  text-transform: uppercase;
  width: 100%;
  padding: 1em 0;
}

.c12:hover {
  opacity: 0.8;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c4 {
  font-size: 1em;
  line-height: 150%;
}

.c2 {
  margin-bottom: 1em;
}

.c3 {
  text-transform: none;
  font-weight: bold;
  font-size: 2.4em;
  line-height: 110%;
}

.c0 {
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  width: 450px;
  margin-top: 3em;
}

.c1 > .c11 {
  margin-top: 1em;
}

.c5 {
  width: 150px;
  height: 150px;
  object-fit: cover;
  display: block;
  margin: 1em auto;
  border-radius: 7px;
}

@media (max-width:768px) {
  .c1 {
    width: 100%;
    min-width: 0;
    max-width: 100%;
  }
}

<div
  class="c0"
>
  <form
    class="c1"
  >
    <div
      class="c2"
    >
      <h2
        class="c3"
      >
        Editar Post
      </h2>
      <p
        class="c4"
      >
        Revise e atualize o seu post
      </p>
    </div>
    <p
      class="c4"
    >
      <strong>
        Preview:
      </strong>
    </p>
    <div
      class=""
    >
      <img
        alt="post title"
        class="c5"
        src="example.png"
      />
    </div>
    <div
      class="c6"
    >
      <label
        class="c7"
        for="image"
      >
        Imagem do Post:
      </label>
      <input
        class="c8"
        id="image"
        name="image"
        placeholder=""
        type="file"
        value=""
      />
    </div>
    <div
      class="c6"
    >
      <label
        class="c7"
        for="title"
      >
        Título do post:
      </label>
      <input
        class="c8"
        id="title"
        name="title"
        placeholder="Título do post"
        type="text"
        value="post title"
      />
    </div>
    <div
      class="c9"
    >
      <span
        class="c10"
      >
        Conteúdo do Post:
      </span>
      <div
        class="quill "
      >
        <div
          class="ql-toolbar ql-snow"
        >
          <span
            class="ql-formats"
          >
            <span
              class="ql-header ql-picker"
            >
              <span
                aria-controls="ql-picker-options-2"
                aria-expanded="false"
                class="ql-picker-label"
                role="button"
                tabindex="0"
              >
                <svg
                  viewBox="0 0 18 18"
                >
                   
                  <polygon
                    class="ql-stroke"
                    points="7 11 9 13 11 11 7 11"
                  />
                   
                  <polygon
                    class="ql-stroke"
                    points="7 7 9 5 11 7 7 7"
                  />
                   
                </svg>
              </span>
              <span
                aria-hidden="true"
                class="ql-picker-options"
                id="ql-picker-options-2"
                tabindex="-1"
              >
                <span
                  class="ql-picker-item"
                  data-value="1"
                  role="button"
                  tabindex="0"
                />
                <span
                  class="ql-picker-item"
                  data-value="2"
                  role="button"
                  tabindex="0"
                />
                <span
                  class="ql-picker-item"
                  data-value="3"
                  role="button"
                  tabindex="0"
                />
                <span
                  class="ql-picker-item"
                  role="button"
                  tabindex="0"
                />
              </span>
            </span>
            <select
              class="ql-header"
              style="display: none;"
            >
              <option
                value="1"
              />
              <option
                value="2"
              />
              <option
                value="3"
              />
              <option
                selected="selected"
              />
            </select>
          </span>
          <span
            class="ql-formats"
          >
            <button
              class="ql-bold"
              type="button"
            >
              <svg
                viewBox="0 0 18 18"
              >
                 
                <path
                  class="ql-stroke"
                  d="M5,4H9.5A2.5,2.5,0,0,1,12,6.5v0A2.5,2.5,0,0,1,9.5,9H5A0,0,0,0,1,5,9V4A0,0,0,0,1,5,4Z"
                />
                 
                <path
                  class="ql-stroke"
                  d="M5,9h5.5A2.5,2.5,0,0,1,13,11.5v0A2.5,2.5,0,0,1,10.5,14H5a0,0,0,0,1,0,0V9A0,0,0,0,1,5,9Z"
                />
                 
              </svg>
            </button>
            <button
              class="ql-italic"
              type="button"
            >
              <svg
                viewBox="0 0 18 18"
              >
                 
                <line
                  class="ql-stroke"
                  x1="7"
                  x2="13"
                  y1="4"
                  y2="4"
                />
                 
                <line
                  class="ql-stroke"
                  x1="5"
                  x2="11"
                  y1="14"
                  y2="14"
                />
                 
                <line
                  class="ql-stroke"
                  x1="8"
                  x2="10"
                  y1="14"
                  y2="4"
                />
                 
              </svg>
            </button>
            <button
              class="ql-underline"
              type="button"
            >
              <svg
                viewBox="0 0 18 18"
              >
                 
                <path
                  class="ql-stroke"
                  d="M5,3V9a4.012,4.012,0,0,0,4,4H9a4.012,4.012,0,0,0,4-4V3"
                />
                 
                <rect
                  class="ql-fill"
                  height="1"
                  rx="0.5"
                  ry="0.5"
                  width="12"
                  x="3"
                  y="15"
                />
                 
              </svg>
            </button>
            <button
              class="ql-link"
              type="button"
            >
              <svg
                viewBox="0 0 18 18"
              >
                 
                <line
                  class="ql-stroke"
                  x1="7"
                  x2="11"
                  y1="7"
                  y2="11"
                />
                 
                <path
                  class="ql-even ql-stroke"
                  d="M8.9,4.577a3.476,3.476,0,0,1,.36,4.679A3.476,3.476,0,0,1,4.577,8.9C3.185,7.5,2.035,6.4,4.217,4.217S7.5,3.185,8.9,4.577Z"
                />
                 
                <path
                  class="ql-even ql-stroke"
                  d="M13.423,9.1a3.476,3.476,0,0,0-4.679-.36,3.476,3.476,0,0,0,.36,4.679c1.392,1.392,2.5,2.542,4.679.36S14.815,10.5,13.423,9.1Z"
                />
                 
              </svg>
            </button>
          </span>
          <span
            class="ql-formats"
          >
            <button
              class="ql-list"
              type="button"
              value="ordered"
            >
              <svg
                viewBox="0 0 18 18"
              >
                 
                <line
                  class="ql-stroke"
                  x1="7"
                  x2="15"
                  y1="4"
                  y2="4"
                />
                 
                <line
                  class="ql-stroke"
                  x1="7"
                  x2="15"
                  y1="9"
                  y2="9"
                />
                 
                <line
                  class="ql-stroke"
                  x1="7"
                  x2="15"
                  y1="14"
                  y2="14"
                />
                 
                <line
                  class="ql-stroke ql-thin"
                  x1="2.5"
                  x2="4.5"
                  y1="5.5"
                  y2="5.5"
                />
                 
                <path
                  class="ql-fill"
                  d="M3.5,6A0.5,0.5,0,0,1,3,5.5V3.085l-0.276.138A0.5,0.5,0,0,1,2.053,3c-0.124-.247-0.023-0.324.224-0.447l1-.5A0.5,0.5,0,0,1,4,2.5v3A0.5,0.5,0,0,1,3.5,6Z"
                />
                 
                <path
                  class="ql-stroke ql-thin"
                  d="M4.5,10.5h-2c0-.234,1.85-1.076,1.85-2.234A0.959,0.959,0,0,0,2.5,8.156"
                />
                 
                <path
                  class="ql-stroke ql-thin"
                  d="M2.5,14.846a0.959,0.959,0,0,0,1.85-.109A0.7,0.7,0,0,0,3.75,14a0.688,0.688,0,0,0,.6-0.736,0.959,0.959,0,0,0-1.85-.109"
                />
                 
              </svg>
            </button>
            <button
              class="ql-list"
              type="button"
              value="bullet"
            >
              <svg
                viewBox="0 0 18 18"
              >
                 
                <line
                  class="ql-stroke"
                  x1="6"
                  x2="15"
                  y1="4"
                  y2="4"
                />
                 
                <line
                  class="ql-stroke"
                  x1="6"
                  x2="15"
                  y1="9"
                  y2="9"
                />
                 
                <line
                  class="ql-stroke"
                  x1="6"
                  x2="15"
                  y1="14"
                  y2="14"
                />
                 
                <line
                  class="ql-stroke"
                  x1="3"
                  x2="3"
                  y1="4"
                  y2="4"
                />
                 
                <line
                  class="ql-stroke"
                  x1="3"
                  x2="3"
                  y1="9"
                  y2="9"
                />
                 
                <line
                  class="ql-stroke"
                  x1="3"
                  x2="3"
                  y1="14"
                  y2="14"
                />
                 
              </svg>
            </button>
          </span>
          <span
            class="ql-formats"
          >
            <button
              class="ql-clean"
              type="button"
            >
              <svg
                class=""
                viewBox="0 0 18 18"
              >
                 
                <line
                  class="ql-stroke"
                  x1="5"
                  x2="13"
                  y1="3"
                  y2="3"
                />
                 
                <line
                  class="ql-stroke"
                  x1="6"
                  x2="9.35"
                  y1="12"
                  y2="3"
                />
                 
                <line
                  class="ql-stroke"
                  x1="11"
                  x2="15"
                  y1="11"
                  y2="15"
                />
                 
                <line
                  class="ql-stroke"
                  x1="15"
                  x2="11"
                  y1="11"
                  y2="15"
                />
                 
                <rect
                  class="ql-fill"
                  height="1"
                  rx="0.5"
                  ry="0.5"
                  width="7"
                  x="2"
                  y="14"
                />
                 
              </svg>
            </button>
          </span>
        </div>
        <div
          class="ql-container ql-snow"
        >
          <div
            class="ql-editor"
            contenteditable="true"
            data-gramm="false"
          >
            <p>
              post content
            </p>
          </div>
          <div
            class="ql-clipboard"
            contenteditable="true"
            tabindex="-1"
          />
          <div
            class="ql-tooltip ql-hidden"
          >
            <a
              class="ql-preview"
              href="about:blank"
              rel="noopener noreferrer"
              target="_blank"
            />
            <input
              data-formula="e=mc^2"
              data-link="https://quilljs.com"
              data-video="Embed URL"
              type="text"
            />
            <a
              class="ql-action"
            />
            <a
              class="ql-remove"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="c6"
    >
      <label
        class="c7"
        for="tags"
      >
        Tags do post:
      </label>
      <input
        class="c8"
        id="tags"
        name="tags"
        placeholder="Adicionar tags ex: #tag1 #tag2"
        type="text"
        value="#tag1 #tag2"
      />
    </div>
    <button
      class="c11 c12"
      type="submit"
    >
      Edite seu post
    </button>
  </form>
</div>
`;
